newspapers.com.js

```diff
@@ -2,103 +2,160 @@
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2017 Peter Binkley
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
 function detectWeb(doc, url) {  
 	return "newspaperArticle";
 }
-function doWeb(doc, url) { 
-	var namespace = doc.documentElement.namespaceURI;
-	var nsResolver = namespace ? function(prefix) {
-		if (prefix == 'x') return namespace; else return null;
-	} : null;
 
+function doWeb(doc, url) { 
 	var newItem = new Zotero.Item("newspaperArticle");
-	var metaArr = new Object();
+	var metaArr = {};
 	var metaTags = doc.getElementsByTagName("meta");
 	for (var i = 0 ; i < metaTags.length ; i++) {
-		metaArr[metaTags[i].getAttribute("property")] = metaTags[i].getAttribute("content");
+		if (metaTags[i].getAttribute("property")) {
+			metaArr[metaTags[i].getAttribute("property")] = metaTags[i].getAttribute("content");
+		}
 	}
-	/*
-	<meta property="og:title" content="Oct 11 1939 Detroit Free Press bork - on Newspapers.com">
-	*/
-	newItem.title = metaArr["og:title"].replace(/(.+?) -( on)? Newspapers.com/, "$1");
+	newItem.title = ZU.xpathText(doc, "//h1[1]");
 	newItem.url = metaArr["og:url"];
 	
+	/*
+		The user can append the author to the title with a forward slash
+		e.g. "My Day / Eleanor Roosevelt"
+	*/
 	if (newItem.title.indexOf('/') >= 0) {
 		var tokens = newItem.title.split("/");
 		var author = tokens[1];
 		newItem.title = tokens[0].trim();
-	    if (author.match("; ")) {
-	      var authors = author.split("; ");
-	      for (var i in authors) {
-	        newItem.creators.push(Zotero.Utilities.cleanAuthor(authors[i], "author"));
-	      }
-	    } else {
-	      newItem.creators.push(Zotero.Utilities.cleanAuthor(author, "author"));
-	    }
+		// multiple authors are separated with semicolons
+		if (author.match("; ")) {
+		  var authors = author.split("; ");
+		  for (var i in authors) {
+			newItem.creators.push(Zotero.Utilities.cleanAuthor(authors[i], "author"));
+		  }
+		} else {
```

Comment:
Actually, you don't need the conditionals here IMO. The `split` function will also work when there is no delimiter `;` present and in this case just return a one-dimensional array. Thus, the first case will work always.